map/src/main/kotlin/edu/artic/map/MapActivity.kt

```diff
@@ -1,22 +1,121 @@
 package edu.artic.map
 
+import android.content.ComponentName
+import android.content.Intent
+import android.content.ServiceConnection
 import android.os.Bundle
-import edu.artic.navigation.NavigationSelectListener
+import android.os.IBinder
+import android.view.View
+import com.fuzz.rx.bindToMain
+import com.fuzz.rx.disposedBy
+import com.jakewharton.rxbinding2.view.visibility
 import edu.artic.base.utils.disableShiftMode
+import edu.artic.media.audio.AudioPlayerService
+import edu.artic.navigation.NavigationSelectListener
 import edu.artic.ui.BaseActivity
 import kotlinx.android.synthetic.main.activity_map.*
+import kotlinx.android.synthetic.main.minimal_exo_playback_control_view.view.*
 
+/**
+ * One of the four primary sections of the app. This Activity always hosts a [MapFragment].
+ *
+ * As a primary section, this class always binds and unbinds with the
+ * background [AudioPlayerService] in [onResume] and [onPause] (respectively
+ * speaking; see the linked audio player documentation for details).
+ */
 class MapActivity : BaseActivity() {
 
     override val layoutResId: Int
         get() = R.layout.activity_map
 
+    var boundService: AudioPlayerService? = null
+    private var audioIntent: Intent? = null
+
+    private val serviceConnection = object : ServiceConnection {
+        override fun onServiceDisconnected(name: ComponentName?) {
+            boundService = null
+        }
+
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            val binder = service as AudioPlayerService.AudioPlayerServiceBinder
+            boundService = binder.getService()
+            setUpAudioServiceBindings()
+        }
+    }
+
+    private fun setUpAudioServiceBindings() {
```

Comment:
this is duplicated in another activity. lets A. Make this a fragment in the UI that can do all the work we need. Or B. extract into a helper viewmodel class or something of that sort to do the logic.